google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/SchemaCompact.java

```diff
@@ -76,24 +101,452 @@ private TableId getTableId(String tableName) {
+        || field == Descriptors.FieldDescriptor.Type.SFIXED64) {
+
+      return true;
+    }
+    return false;
+  }
+
+  private static boolean isCompatibleWithBQTimestamp(Descriptors.FieldDescriptor.Type field) {
+    if (isCompatibleWithBQInteger(field)) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Checks if proto field option is compatible with BQ field mode.
+   *
+   * @param protoField
+   * @param BQField
+   * @param protoScope Debugging purposes to show error if messages are nested.
+   * @param BQScope Debugging purposes to show error if messages are nested.
+   * @throws IllegalArgumentException if proto field type is incompatible with BQ field type.
+   */
+  private void protoFieldModeIsCompatibleWithBQFieldMode(
+      Descriptors.FieldDescriptor protoField, Field BQField, String protoScope, String BQScope)
+      throws IllegalArgumentException {
+    if (BQField.getMode() == null) {
+      throw new IllegalArgumentException(
+          "Big query schema contains invalid field option for " + BQScope + ".");
+    }
+    switch (BQField.getMode()) {
+      case REPEATED:
+        if (!protoField.isRepeated()) {
+          throw new IllegalArgumentException(
+              "Given proto field "
+                  + protoScope
+                  + " is not repeated but Big Query field "
+                  + BQScope
+                  + " is.");
+        }
+        break;
+      case REQUIRED:
+        if (!protoField.isRequired()) {
+          throw new IllegalArgumentException(
+              "Given proto field "
+                  + protoScope
+                  + " is not required but Big Query field "
+                  + BQScope
+                  + " is.");
+        }
+        break;
+      case NULLABLE:
+        if (protoField.isRepeated()) {
+          throw new IllegalArgumentException(
+              "Given proto field "
+                  + protoScope
+                  + " is repeated but Big Query field "
+                  + BQScope
+                  + " is optional.");
+        }
+        break;
+    }
+  }
+  /**
+   * Checks if proto field type is compatible with BQ field type.
+   *
+   * @param protoField
+   * @param BQField
+   * @param allowUnknownFields
+   * @param protoScope Debugging purposes to show error if messages are nested.
+   * @param BQScope Debugging purposes to show error if messages are nested.
+   * @throws IllegalArgumentException if proto field type is incompatible with BQ field type.
+   */
+  private void protoFieldTypeIsCompatibleWithBQFieldType(
+      Descriptors.FieldDescriptor protoField,
+      Field BQField,
+      boolean allowUnknownFields,
+      String protoScope,
+      String BQScope)
+      throws IllegalArgumentException {
+
+    LegacySQLTypeName BQType = BQField.getType();
+    Descriptors.FieldDescriptor.Type protoType = protoField.getType();
+
+    boolean match = false;
+
+    if (BQType == LegacySQLTypeName.BOOLEAN) {
```

Comment:
I originally tried using switch statements on LegacySQLTypeName, but I got a compile error because switch statements can only be used on primitives and not objects so I thought that I should just use if statements. I could however do a switch statement on the string version of it using the toString() method.

Reply:
Yeah, try switch on hashcode or value.

Reply:
Got it, I used the string value.